src/documents/firstDiscoveryServer.md

```diff
@@ -0,0 +1,383 @@
+
+Access to the GPII Preferences Server is mediated by the [GPII OAuth2 Security layer](https://wiki.gpii.net/w/GPII_OAuth_2_Guide). The First Discovery Server communicates with the Security layer via the [Client Credentials Grant](https://wiki.gpii.net/w/GPII_OAuth_2_Guide#Client_Credentials_Grant) workflow.
+
+
+## Configuration ##
+
+The First Discovery Server can be configured via [Kettle Configs](https://github.com/amb26/kettle/blob/KETTLE-32/README.md#structure-of-a-kettle-config). A set of these are provided with the server in the [config](https://github.com/GPII/first-discovery-server/tree/master/src/config) directory. An integrator can choose to make use of the provided configs directly or use them as the basis for creating new configs.
+
+The [`gpii.firstDiscovery.server.configurator`](https://github.com/GPII/first-discovery-server/tree/master/src/js/firstDiscoveryServer.js) grade defines a default schema which the configuration is validated against. If the validation fails, the application will throw and error.
+
+### environment.json ###
+
+Accepts environment variables for all of the critical configuration options that are needed for running the First Discovery Server and connecting to the OAuth2 security server.
+
+(see: [environment.json](https://github.com/GPII/first-discovery-server/tree/master/src/config/environment.json))
+
+<table>
+    <thead>
+        <tr>
+            <th>Environment Variable</th>
+            <th>Description</th>
+        <tr>
+    </thead>
+    <tbody>
+        <tr>
+            <td>
+                <code>FIRST_DISCOVERY_SERVER_TCP_PORT</code>
+            </td>
+            <td>
+                The port number that the First Discovery Server is hosted at.
+                <p>
+                    Example:
+                    "8088"
+                </p>
+                <p>
+                    Option Path:
+                    <code>"port"</code>
+                </p>
+            </td>
+        </tr>
+        <tr>
+            <td>
+                <code>GPII_OAUTH2_TCP_PORT</code>
+            </td>
+            <td>
+                The port number that the GPII OAuth2 server is hosted at.
+                <p>
+                    Example:
+                    "8081"
+                </p>
+                <p>
+                    Option Path:
+                    <code>"preferencesConfig.securityServer.port"</code>
```

Comment:
I think it's confusing to have quotes around these strings - which suggest that the environment variable value itself needs to have quotes in it. Since these are an external part of the shell process, strings are the only type supported.